diff --git a/python/dolfinx/__init__.py b/python/dolfinx/__init__.py
index 52edc9c970..e859c3e5de 100644
--- a/python/dolfinx/__init__.py
+++ b/python/dolfinx/__init__.py
@@ -35,6 +35,18 @@ except ImportError:
     default_scalar_type = _np.float64
     default_real_type = _np.float64
 
+import prevent_user_site_imports
+prevent_user_site_imports.check(
+    "dolfinx", "https://fenicsproject.discourse.group/",
+    "/usr/lib/python3/dist-packages",
+    ["basix", "ffcx", "ufl"],
+    ["python3-basix", "python3-ffcx", "python3-ufl"],
+    ["fenics-basix", "fenics-ffcx", "fenics-ufl"],
+    [False, False, False],
+    ["", "", ""]
+)
+del prevent_user_site_imports
+
 from dolfinx import common
 from dolfinx import cpp as _cpp
 from dolfinx import fem, geometry, graph, io, jit, la, log, mesh, nls, plot
