diff --git a/dolfinx/__init__.py b/dolfinx/__init__.py
index 52edc9c..b90a2d2 100644
--- a/dolfinx/__init__.py
+++ b/dolfinx/__init__.py
@@ -35,6 +35,8 @@ except ImportError:
     default_scalar_type = _np.float64
     default_real_type = _np.float64
 
+import dolfinx._deb_consistency_check
+
 from dolfinx import common
 from dolfinx import cpp as _cpp
 from dolfinx import fem, geometry, graph, io, jit, la, log, mesh, nls, plot
