diff --git a/python/dolfin/__init__.py b/python/dolfin/__init__.py
index fcb52d131..f436ba028 100644
--- a/python/dolfin/__init__.py
+++ b/python/dolfin/__init__.py
@@ -30,6 +30,20 @@ del sys
 # sys.setdlopenflags(stored_dlopen_flags)
 # del sys

+import pusimp
+pusimp.prevent_user_site_imports(
+    "dolfin", "apt", "https://fenicsproject.discourse.group/",
+    "/usr/lib/python3/dist-packages",
+    ["dijitso", "ffc", "FIAT", "ufl_legacy", "ufl"],
+    ["fenics-dijitso", "fenics-ffc", "fenics-fiat", "fenics-ufl-legacy", "fenics-ufl"],
+    [False, False, False, False, True],
+    ["", "", "", "", (
+        " Be aware that legacy dolfin codes must now import ufl_legacy instead of ufl, "
+        "see https://fenicsproject.discourse.group/t/announcement-ufl-legacy-and-legacy-dolfin/11583 ."
+    )]
+)
+del pusimp
+
 # Import cpp modules
 from .cpp import __version__

